standard-cluster-services/src/main/java/org/terracotta/monitoring/PlatformService.java

```diff
@@ -54,4 +54,38 @@
    * Provides the raw config file from server startup
    */
   InputStream getPlatformConfiguration();
+
+  /**
+   * Stops the server if it is in any of the passive states
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt (exit with correct exit codes).
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatformIfPassive(RestartMode restartMode);
```

Comment:
should have a boolean return or throw a checked exception in the case that server is not in the proper state to shutdown

Reply:
I think it needs to be a checked exception so that it is handled in the context which it is thrown.

Reply:
added checked exception